server/src/main/java/io/spine/server/BoundedContext.java

```diff
@@ -350,7 +350,31 @@ private void registerIfAware(Object contextPart) {
     }
 
     /**
-     * Finds a repository by the state class of entities.
+     * Attempts to find a repository by the state class of entities.
+     *
+     * <p>Returns {@link Optional#empty()} if:
+     * <ol>
+     *     <li>A repository for the given entity class is not registered in this context.
+     *     <li>The requested entity is {@linkplain Visibility#NONE not visible}.
+     * </ol>
+     *
+     * @param entityStateClass
+     *         the class of the state of the entity managed by the resulting repository
+     * @return the requested repository or {@link Optional#empty()} if it's not found or not
+     *         visible
+     * @see VisibilityGuard
+     */
+    @Internal
+    public Optional<Repository> findRepository(Class<? extends Message> entityStateClass) {
```

Comment:
Maybe we could have a method telling whether the repository is registered? In this way, the API would be much more consistent. 

The`repositoryFor` is very strict and assumes the repository *is* registered, but there is no way to tell whether it is. So I would fill this gap instead of duplicating the API call with a very small lexical difference.